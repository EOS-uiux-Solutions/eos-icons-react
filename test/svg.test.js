/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'
import { Eos10kFilled, Eos10kOutlined, EosloadingAnimated } from '../lib'

test('React Filled SVG Component test', () => {
  const component = renderer.create(
    <Eos10kFilled/>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('React Filled SVG Component Prop Test', () => {
  const component = renderer.create(
    <Eos10kFilled color="blue" rotate={90} size="xxxl" verticalFlip={true} horizontalFlip={true}/>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('React Outlined SVG Component test', () => {
  const component = renderer.create(
    <Eos10kOutlined/>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('React Outlined SVG Component Prop Test', () => {
  const component = renderer.create(
    <Eos10kOutlined color="blue" rotate={90} size="xxxl" verticalFlip={true} horizontalFlip={true}/>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('React animated SVG Component test', () => {
  const component = renderer.create(
    <EosloadingAnimated/>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('React animated SVG Component Prop Test', () => {
  const component = renderer.create(
    <EosloadingAnimated color="blue" rotate={90} size="xxxl" verticalFlip={true} horizontalFlip={true}/>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
