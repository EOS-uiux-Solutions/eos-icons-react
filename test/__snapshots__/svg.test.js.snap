// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Filled SVG Component Prop Test 1`] = `
<svg
  fill="blue"
  height={64}
  transform="rotate(90) translate(24, 24) scale(-1, -1)"
  viewBox="0 0 24 24"
  width={64}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M10 10.5h1.5v3H10zM19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM7.5 15H6v-4.5H4.5V9h3v6zm5.5-1c0 .55-.45 1-1 1H9.5c-.55 0-1-.45-1-1v-4c0-.55.45-1 1-1H12c.55 0 1 .45 1 1v4zm6.5 1h-1.75L16 12.75V15h-1.5V9H16v2.25L17.75 9h1.75l-2.25 3 2.25 3z"
  />
</svg>
`;

exports[`React Filled SVG Component test 1`] = `
<svg
  fill="black"
  height={18}
  transform="rotate(0) translate(0, 0) scale(1, 1)"
  viewBox="0 0 24 24"
  width={18}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M10 10.5h1.5v3H10zM19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM7.5 15H6v-4.5H4.5V9h3v6zm5.5-1c0 .55-.45 1-1 1H9.5c-.55 0-1-.45-1-1v-4c0-.55.45-1 1-1H12c.55 0 1 .45 1 1v4zm6.5 1h-1.75L16 12.75V15h-1.5V9H16v2.25L17.75 9h1.75l-2.25 3 2.25 3z"
  />
</svg>
`;

exports[`React Outlined SVG Component Prop Test 1`] = `
<svg
  fill="blue"
  height={64}
  transform="rotate(90) translate(24, 24) scale(-1, -1)"
  viewBox="0 0 24 24"
  width={64}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M19,3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5C21,3.9,20.1,3,19,3z M19,9v6v4H5v-8.5h1V15h1.5V9H5V5 h14V9z"
  />
  <polygon
    points="15.5,11.25 15.5,9 14,9 14,15 15.5,15 15.5,12.75 17.25,15 19,15 16.75,12 19,9 17.25,9"
  />
  <path
    d="M9.5,15H12c0.55,0,1-0.45,1-1v-4c0-0.55-0.45-1-1-1H9.5c-0.55,0-1,0.45-1,1v4C8.5,14.55,8.95,15,9.5,15z M10,10.5h1.5v3 H10V10.5z"
  />
</svg>
`;

exports[`React Outlined SVG Component test 1`] = `
<svg
  fill="black"
  height={18}
  transform="rotate(0) translate(0, 0) scale(1, 1)"
  viewBox="0 0 24 24"
  width={18}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M19,3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5C21,3.9,20.1,3,19,3z M19,9v6v4H5v-8.5h1V15h1.5V9H5V5 h14V9z"
  />
  <polygon
    points="15.5,11.25 15.5,9 14,9 14,15 15.5,15 15.5,12.75 17.25,15 19,15 16.75,12 19,9 17.25,9"
  />
  <path
    d="M9.5,15H12c0.55,0,1-0.45,1-1v-4c0-0.55-0.45-1-1-1H9.5c-0.55,0-1,0.45-1,1v4C8.5,14.55,8.95,15,9.5,15z M10,10.5h1.5v3 H10V10.5z"
  />
</svg>
`;

exports[`React animated SVG Component Prop Test 1`] = `
<svg
  fill="blue"
  height={64}
  transform="rotate(90) translate(24, 24) scale(-1, -1)"
  viewBox="0 0 24 24"
  width={64}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
    opacity=".5"
  />
  <path
    d="M20,12h2A10,10,0,0,0,12,2V4A8,8,0,0,1,20,12Z"
  >
    <animateTransform
      attributeName="transform"
      dur="1s"
      from="0 12 12"
      repeatCount="indefinite"
      to="360 12 12"
      type="rotate"
    />
  </path>
</svg>
`;

exports[`React animated SVG Component test 1`] = `
<svg
  fill="black"
  height={18}
  transform="rotate(0) translate(0, 0) scale(1, 1)"
  viewBox="0 0 24 24"
  width={18}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
    opacity=".5"
  />
  <path
    d="M20,12h2A10,10,0,0,0,12,2V4A8,8,0,0,1,20,12Z"
  >
    <animateTransform
      attributeName="transform"
      dur="1s"
      from="0 12 12"
      repeatCount="indefinite"
      to="360 12 12"
      type="rotate"
    />
  </path>
</svg>
`;
